packages/itwinui-react/src/core/Header/HeaderLogo.tsx

```diff
@@ -60,8 +65,8 @@ export const HeaderLogo = (props: HeaderLogoProps) => {
           })
         : undefined}
       {children && <span className='iui-header-brand-label'>{children}</span>}
-    </div>
+    </Element>
   );
-};
+}) as PolymorphicForwardRefComponent<'div', HeaderLogoOwnProps>;
```

Comment:
What does this mean or do? Why does it have type 'div' in there?

Reply:
You can check out the comments at: https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-react/src/core/utils/props.ts#L42-L53

It's advanced typescript stuff and it's a bit hard to explain tbh. It is a generic type (think of it like a function) which takes two parameters - the first param is the default element name (so we want it to have all `'div'` props by default) and the second param is the component's own props (e.g. `logo`) to be added to the div props.

The result of this is that you get correctly typed and autocompletable attributes. For example, when I use `as='a'`, it automatically knows about [`href`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href).

![](https://user-images.githubusercontent.com/9084735/223140642-9ac81d87-96aa-42dc-a43e-6ceffcc5705e.png)
![](https://user-images.githubusercontent.com/9084735/223140738-fcb12cc7-d311-4ffd-94f0-924dcc2a1f4b.png)

You can also ignore all the typescript stuff and review only the javascript parts - the code is identical to what I suggested in https://github.com/iTwin/iTwinUI/issues/920#issuecomment-1362952436.

Reply:
Thanks. That's helpful